packages/pluggableWidgets/progress-circle-native/tests/e2e/ProgressCircle.spec.ts

```diff
@@ -0,0 +1,72 @@
+import { device, expect, waitFor } from "detox";
+import { Widget } from "../../../../../tests/e2e";
+
+describe("Progress Circle", () => {
+    beforeAll(async () => {
+        const progressCircleWidget = Widget("btnProgressCircle").getElement();
+        await progressCircleWidget.tap();
+
+        const textBox = Widget("textBoxProgressCircleValue").getElement();
+        await waitFor(textBox).toBeVisible().withTimeout(10000);
+        await textBox.tap();
+        await textBox.clearText();
+        await textBox.typeText("75");
+    });
+
+    it("renders the progress circle with text as percentage", async () => {
+        const percentageText = "75%";
+        const progressCircleId = Widget("progressCirclePercentageText");
+        const progressCircle = progressCircleId.getElement();
+        const progressCircleText = progressCircleId.getText(percentageText);
+
+        await expect(progressCircle).toBeVisible();
+        await progressCircle.tap();
+
+        await expect(progressCircleText).toBeVisible();
+        await expect(progressCircleText).toHaveText(percentageText);
+    });
+
+    it("renders the progress circle with text as custom", async () => {
+        const customText = "75/100";
+        const progressCircleId = Widget("progressCircleCustomText");
+        const progressCircle = progressCircleId.getElement();
+        const progressCircleText = progressCircleId.getText(customText);
+
+        await expect(progressCircle).toBeVisible();
+        await progressCircle.tap();
+
+        await expect(progressCircleText).toBeVisible();
+        await expect(progressCircleText).toHaveText(customText);
+    });
+
+    // To-do: Once detox releases the method to assert empty string
```

Comment:
nit: should be either todo or TODO; easier to find when code finding

also, maybe make a JIRA for this?

Reply:
Good one! I've made the required change and also created a JIRA story NC-456 for it.